api/src/modules/renown/kyc-service.ts

```diff
@@ -0,0 +1,201 @@
+      verifiableCredentialEIP712Index: {
+        edges: { node: CeramicPowerhouseVerifiableCredential }[];
+      };
+    }>(fetchEntries, {
+      first,
+      skip,
+      input: {
+        and: [
+          {
+            or: [
+              {
+                where: {
+                  revocationDate: {
+                    greaterThan: date,
+                  },
+                },
+              },
+              {
+                where: {
+                  revocationDate: {
+                    isNull: true,
+                  },
+                },
+              },
+            ],
+          },
+        ],
+      },
+    });
+
+    return entries as any[];
+  }
+
+  #fetchEntryFromCeramic = async (issuerId: string, subjectId: string) => {
+    const date = new Date().toISOString()
+    const result = await this.client.executeQuery<{
+      verifiableCredentialEIP712Index: {
+        edges: { node: CeramicPowerhouseVerifiableCredential }[];
+      };
+    }>(fetchEntries, {
+      input: {
+        and: [
+          {
+            where: {
+              issuerId: {
+                equalTo: issuerId,
+              },
+              subjectId: {
+                equalTo: subjectId,
+              },
+            },
+          },
+          {
+            or: [
+              {
+                where: {
+                  revocationDate: {
+                    greaterThan: date,
+                  },
+                },
+              },
+              {
+                where: {
+                  revocationDate: {
+                    isNull: true,
+                  },
+                },
+              },
+            ],
+          },
+        ],
+      },
+    });
+
+    if (result.errors) {
+      return null;
+    }
+
+    const { data } = result;
+    if (!data || data.verifiableCredentialEIP712Index.edges.length === 0) {
+      return null;
+    }
+
+    return data.verifiableCredentialEIP712Index.edges[0]!.node;
+  }
+
+  #updateEntries = async () => {
+    const first = 100;
+    let skip = 0;
+    let entries: any[] = [];
+    let foundEntries = 0;
+    const fetchEntriesLoop = async () => {
+      const newEntries = await this.#fetchEntriesFromCeramic(first, skip);
+      foundEntries = newEntries.length;
+      if (foundEntries === first) {
+        skip += first;
+        entries = entries.concat(newEntries);
+        await fetchEntriesLoop();
+      } else {
+        entries = entries.concat(newEntries);
+      }
+    };
+
+    await fetchEntriesLoop();
+    this.entries = entries;
+  }
+
+  #updateEntry = async (issuerId: string, subjectId: string) => {
+    const entry = await this.#fetchEntryFromCeramic(issuerId, subjectId);
+    const index = this.entries.findIndex((e) => e.issuerId === issuerId && e.subjectId === subjectId);
+    if (!entry && index !== -1) {
+      delete this.entries[index];
+      return null;
+    }
+
+    if (index !== -1) {
+      this.entries[index] = entry;
```

Comment:
Same here, if we a different credential stored locally then it should be deleted. Probably the expiration date was reached.

Reply:
I just check here whether we store that credential already. if this is the case I overwrite it otherwise I'll add it.
I lookup the index of the entry bei issuerId and subjectId. this means even if there is a new credential. it should overwrite the existing one.